infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/PodMerger.java

```diff
@@ -141,4 +154,33 @@ private Deployment createEmptyDeployment(String name) {
         .endSpec()
         .build();
   }
+
+  private PodSecurityContext mergeSecurityContexts(
+      @Nullable PodSecurityContext a, @Nullable PodSecurityContext b) throws ValidationException {
+    return nonNullOrEqual(a, b, "Cannot merge pods with different security contexts: %s, %s");
+  }
+
+  private String mergeServiceAccount(@Nullable String a, @Nullable String b)
+      throws ValidationException {
+    return nonNullOrEqual(a, b, "Cannot merge pods with different service accounts: %s, %s");
+  }
+
+  private String mergeServiceAccountName(@Nullable String a, @Nullable String b)
+      throws ValidationException {
+    return nonNullOrEqual(a, b, "Cannot merge pods with different service account names: %s, %s");
+  }
+
+  @Nullable
+  private static <T> T nonNullOrEqual(@Nullable T a, @Nullable T b, String errorMessageTemplate)
+      throws ValidationException {
+    if (a == null) {
+      return b;
```

Comment:
can potentially return `null` of both are. is that ok ?

Reply:
It can be null, as suggested by the `@Nullable` on the method.

Reply:
ok method name starts with `notNull` so little confusing